### DISCUSSION TITLE
Cannot embed the library to project

### DISCUSSION THREAD

**Discussion by k2465** [roerohan/react-vnc#35] [Q&A] [answered]
Hello! First of all thank you so much for this useful integration library for React :)
I'm having a little issue embedding this in my project.
What I did is just `npm install react-vnc` and then pasted the code in the README

```
import React from 'react';
import { VncScreen } from 'react-vnc';

function App() {
  return (
    <VncScreen
      url='ws://your-vnc-url.com'
      scaleViewport
      background="#000000"
      style={{
        width: '75vw',
        height: '75vh',
      }}
      ref={ref}
    />
  );
}

export default App;
```

But I'm getting this:

![Screenshot_20220518_131302](https://user-images.githubusercontent.com/105788860/169026682-2bd326a3-18ca-4091-a6f9-a4b1f1d0cf0c.png)

I did a lot of tests with the package.json file, I did build the library and tried to embed it, plus a lot of other tests but I didn't get any result.
I'm sure I'm missing something stupid but I'm pretty much lost with no other solutions :(

Thank you for your time

**Comment by draentropia** [comment]
I had the same issue: ref is missing. See my component to get more info:

`import React, { useRef } from 'react'
import { VncScreen } from 'react-vnc'

const Terminal = () => {

  const vncScreenRef = useRef(VncScreen)

  return (
    <>
    <div>
    <h1>Terminal</h1>

    <VncScreen
      url='ws://192.168.60.111:5900'
      scaleViewport
      background="#000000"
      style={{
        width: '75vw',
        height: '75vh',
      }}
      ref={vncScreenRef}
    />
    </div>
    </>
  )
}

export default Terminal
`

**Reply by roerohan** [reply]
Hey @k2465 @draentropia 
Could you tell me which React version you're using? Maybe there is an issue with `React 18`, they made a bunch of breaking changes. I haven't tried running it with React 18 myself. Try running it on React 17, it should work as intended.

**Reply by roerohan** [reply]
Hey, this does not seem to be a `React 18` issue. The library works fine even with React 18. Can you change the `ref` that you've created from

```ts
const vncScreenRef = useRef(VncScreen)
```

to 

```ts
const vncScreenRef = useRef();
```

I have updated the snippet in the `README` to a working version.

**Comment by roerohan** [comment, marked as answer]
> Hello! First of all thank you so much for this useful integration library for React :) I'm having a little issue embedding this in my project. What I did is just `npm install react-vnc` and then pasted the code in the README
> 
> ```
> import React from 'react';
> import { VncScreen } from 'react-vnc';
> 
> function App() {
>   return (
>     <VncScreen
>       url='ws://your-vnc-url.com'
>       scaleViewport
>       background="#000000"
>       style={{
>         width: '75vw',
>         height: '75vh',
>       }}
>       ref={ref}
>     />
>   );
> }
> 
> export default App;
> ```
> 
> But I'm getting this:
> 
> ![Screenshot_20220518_131302](https://user-images.githubusercontent.com/105788860/169026682-2bd326a3-18ca-4091-a6f9-a4b1f1d0cf0c.png)
> 
> I did a lot of tests with the package.json file, I did build the library and tried to embed it, plus a lot of other tests but I didn't get any result. I'm sure I'm missing something stupid but I'm pretty much lost with no other solutions :(
> 
> Thank you for your time

Hey, so the error in your code snippet was that `ref` was not created. You can use `useRef()` to create a `ref` and pass it to the `VncScreen` component. I've updated the snippet in the library documentation](https://github.com/roerohan/react-vnc#README.md).

**Comment by draentropia** [comment]
Im'm using react 18, but the issue is with react-scripts 5 for sure. I'd
just downgraded to react-scripts 4 and it works like a charm.
